src/api/ImpossibleGoalsApi/index.ts

```diff
@@ -0,0 +1,54 @@
+import { db } from "@src/models";
+import { getGoal } from "../GoalsAPI";
+
+export const addImpossibleGoal = async (impossibleGoalId: string) => {
+  try {
+    const goal = await getGoal(impossibleGoalId);
+    if (!goal) throw new Error("Goal not found.");
```

Comment:
we can't throw error directly like this as it will break the code instead a custom hook can be created that catches error. can you create a custom hook that can be used as a wrapper for all api(s)?

Reply:
Could you please provide some suggestions on how we can create the custom hook you mentioned? I have experience with creating wrapper functions for HTTP API calls but can't come up with logic for doing the same for our API, which directly communicates with the database.

Reply:
the hook will just be a functional component that takes the function in argument and will call it and if that fn throws an error it will catch it & will show the same in toast.

Reply:
@Tushar-4781, I tried but couldn't develop the hook that can work for all our current APIs. For now, we can console log this errors instead of throwing it. We can take care of your concern in the new PR.